### ISSUE TITLE
Cache not cleared after removing `[[tool.mypy.overrides]]` from `pyproject.toml`

### ISSUE THREAD

**Issue by bersbersbers** [python/mypy#17032]
**Bug Report**

A project that relies on `[[tool.mypy.overrides]]` to fully type-check still type-checks after removing these sections, until the cache is invalidated by some other method.

**To Reproduce**

I do not have time to create a test project for reproduction, but the step go along this way:

```
(.venv) C:\Code\project>mypy .
Success: no issues found in 45 source files

(.venv) C:\Code\project>mypy . --no-incremental 
Success: no issues found in 45 source files

(.venv) C:\Code\project>rem remove [[tool.mypy.overrides]] from pyproject.toml

(.venv) C:\Code\project>mypy .     
Success: no issues found in 45 source files

(.venv) C:\Code\project>mypy . --no-incremental
project\util\colormaps.py:6: error: Skipping analyzing "colorcet": module is installed, but missing library stubs or py.typed marker  [import-untyped]
project\util\colormaps.py:6: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
project\util\colormaps.py:6: note: See https://mypy.rtfd.io/en/stable/_refs.html#code-import-untyped for more info
Found 1 error in 1 file (checked 45 source files)
```

**Your Environment**

- Mypy version used: 1.9.0
- Python 3.12.2

**Comment by ilevkivskyi** [comment]
@bersbersbers Can you give some details on how to reproduce this? In particular what was the content of overrides sections that caused issues? Was it `ignore_errors = true`, or something more specific? This kind of bugs is hard to diagnose without some kind of repro.

**Comment by bersbersbers** [comment]
I checked my repository from two years ago and I find (only) this:

```toml
[[tool.mypy.overrides]]
# https://github.com/holoviz/colorcet/
module = "colorcet"
ignore_missing_imports = true

[[tool.mypy.overrides]]
# https://github.com/Suor/funcy/issues/106
module = "funcy"
ignore_missing_imports = true

[[tool.mypy.overrides]]
# https://github.com/pyqtgraph/pyqtgraph/issues/2409
module = "pyqtgraph.*"
ignore_missing_imports = true

[[tool.mypy.overrides]]
# https://github.com/pytest-dev/pytest-qt/issues/417
module = "pytestqt.*"
ignore_missing_imports = true

```

This seems to match my original report:
```
project\util\colormaps.py:6: error: Skipping analyzing "colorcet": module is installed, but missing library stubs or py.typed marker  [import-untyped]
```

If this does not reproduce the error, I could take a look at checking out that old state and retry with that old mypy version - do let me know.

**Comment by ilevkivskyi** [comment]
@bersbersbers Great, thanks for confirming! This confirms my suspicion that this is a duplicate of https://github.com/python/mypy/issues/7777 (which I was going to take a look soon).
